rootio/scanner.go

```diff
@@ -54,11 +54,36 @@ func (s *baseScanner) SeekEntry(i int64) error {
 	if s.err != nil {
 		return s.err
 	}
+	if s.chain {
+		if s.cur >= s.n {
+			itree := s.findTree(i)
+			s.loadTree(itree)
+		}
+	}
 	s.i = i
 	s.cur = i - 1
 	return s.err
 }
 
+// find the tree number in the chain that contains entry i
+func (s *baseScanner) findTree(i int64) int {
+	ch := s.tree.(*tchain)
+	var j int = 0
+	for j <= len(ch.trees) {
```

Comment:
the way it's implemented as of now, `findTree` will return an invalid index (ie `len(ch.trees)`) when the requested entry `i` is outside of the chain.
fine, but I think `-1` would be better and less surprising considering a few of the Go APIs.
_e.g.:_
```
$> go doc strings.Index
func Index(s, substr string) int
    Index returns the index of the first instance of substr in s, or -1 if
    substr is not present in s.
```